tools/ci_connector_ops/ci_connector_ops/qa_checks.py

```diff
@@ -84,6 +83,29 @@ def check_connector_icon_is_available(connector: Connector) -> bool:
     """
     return connector.icon_path.exists()
 
+
+def read_all_files_in_directory(directory: Path, path_patterns_to_ignore: Optional[Set[str]] = None) -> Iterable[Tuple[str, str]]:
+    path_patterns_to_ignore = path_patterns_to_ignore if path_patterns_to_ignore is not None else {}
+    for path in directory.rglob("*"):
+        ignore = bool([pattern for pattern in path_patterns_to_ignore if pattern in str(path)])
```

Comment:
This is a little fragile because the `in` matches too much. In the`PATH_PATTERNS_TO_IGNORE_FOR_HTTPS_CHECKS` set you have `test` which would match `contest.py` or `protest.yml`. [Pathlib has a match method](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.match) which takes a regex which could be helpful here. Something like:
```python
ignore = any([path.match(pattern) for pattern in path_patterns_to_ignore])
```